misc/CreateAssociationTools.kt

```diff
@@ -0,0 +1,109 @@
+            val image = parts[7]
+            val principalEmailAddress = parts[8]
+
+            return Association(
+                uid = uid,
+                url = url,
+                name = name,
+                fullName = fullName,
+                category = category,
+                description = description,
+                followersCount = followersCount,
+                members = emptyList(), //does not need it for the creation
+                roles = emptyList(), //does not need it for the creation
+                image = image,
+                events = Event.emptyFirestoreReferenceList(),
+                principalEmailAddress = principalEmailAddress
+            )
+        } catch (e: Exception) {
+            Log.e("MainActivity", "Error parsing line: $line", e)
+            return null
+        }
+    }
+
+    private fun processAndSaveAssociations(
+        inputStream: InputStream,
+        saveAssociation: (Association, InputStream?) -> Unit,
+        context: Context
+    ) {
+        val bufferedReader = BufferedReader(InputStreamReader(inputStream))
+        try {
+            bufferedReader.useLines { lines ->
+                lines.forEach { line ->
+                    val association = parseLineToAssociation(line)
+                    association?.let {
+
+
+                        val resourceId = context.resources.getIdentifier(it.image, "raw", context.packageName)
+
+                        if (resourceId != 0) {
+                            val imageStream = try {
+                                context.resources.openRawResource(resourceId)
+                            } catch (e: Exception) {
+                                Log.e("MainActivity", "Error opening resource: ${it.image}", e)
+                                null
+                            }
+                            saveAssociation(it, imageStream)
+                        } else {
+                            Log.e("MainActivity", "Resource not found: ${it.image}")
+                        }
+
+
+                    }
+                }
+            }
+        } catch (e: Exception) {
+            Log.e("MainActivity", "Error processing file", e)
+        }
+    }
+
+    // IN ORDER TO MAKE IT WORK, CALL THIS INTO MAINACTIVITY UnioApp function :
+    /*
+    val context = LocalContext.current
+
+    LaunchedEffect(Unit) {
+
+    withContext(Dispatchers.IO) {
+      delay(30000)
+
+      val inputStream = context.resources.openRawResource(R.raw.associations)
+      processAndSaveAssociations(inputStream, { association, imageStream ->
+        associationViewModel.saveAssociation(
+          association,
+          imageStream = imageStream,
+          onSuccess = { Log.i("UnioApp", "Saved association: ${association.name}") },
+          onFailure = { exception ->
+            Log.e("UnioApp", "Failed to save association", exception)
+          }
+        )
+      }, context)
+    }
+  }*/
```

Comment:
Consider not relying on manually adding code into MainActivity. Instead, you could use a Kotlin `main` function that doesn't even rely on running on an emulated device. 

Also, how can this code in MainActivity even work ? By default if users are not authenticated, they cannot add data to the Firestore database (is that why there is a `delay(30000)` ?). 

You should really consider making this a simple Kotlin `main` entry point, and use Firebase service accounts to bypass security rules. That would additionally greatly speed up the execution time.

Reply:
Okay I can see your point and I agree; but is it really useful for this kind of utilization ? 
I will merge it so that everyone can use it to fullfill the database, but will refactor it soon with what you say.